fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt

```diff
@@ -41,12 +67,38 @@ fun DebugOverlay() {
                 .align(Alignment.CenterStart)
                 .padding(start = 16.dp),
             onClick = {
-                scope.launch {
-                    snackbarState.showSnackbar("Show debug drawer")
-                }
+                drawerEnabled = true
             },
         )
 
+        // ModalDrawer utilizes a Surface, which blocks ALL clicks behind it, preventing the app
+        // from being interactable. This cannot be overridden in the Surface API, so we must hide
+        // the entire drawer when it is closed. Whenever the app is Compose from
+        // the root layer, we can remove this.
+        if (drawerEnabled) {
+            val currentLayoutDirection = LocalLayoutDirection.current
+            val sheetLayoutDirection = when(currentLayoutDirection) {
+                LayoutDirection.Rtl -> LayoutDirection.Ltr
+                LayoutDirection.Ltr -> LayoutDirection.Rtl
+            }
+
+            // Force the drawer to always open from the opposite side of the screen
+            CompositionLocalProvider(LocalLayoutDirection provides sheetLayoutDirection) {
+                ModalDrawer(
+                    drawerContent = {
+                        // Ensure the drawer's content still respects the device's direction setting
```

Comment:
Leaving the comments for now, but open to cleaning some/all of these up before landing

Reply:
I don't think this one is necessary or could at least be combined more with the one above `CompositionLocalProvider`. The rest seem useful to me